src/App.jsx

```diff
@@ -1,35 +1,37 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import { BrowserRouter, Routes, Route, Link } from "react-router-dom";
+import Home from "./pages/Home.jsx";
+import Search from "./pages/Search.jsx";
+import Details from "./pages/Details.jsx";
 
-function App() {
-  const [count, setCount] = useState(0)
+const navStyle = {
+  display: "flex",
+  gap: 16,
+  padding: "12px 16px",
+  borderBottom: "1px solid #eee",
+  position: "sticky",
+  top: 0,
+  background: "#fff",
+  zIndex: 10,
+};
 
+export default function App() {
   return (
-    <>
-      <div>
-        <a href="https://vite.dev" target="_blank">
-          <img src={viteLogo} className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://react.dev" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.jsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-    </>
-  )
-}
+    <BrowserRouter>
+      {/* Dev Nav*/}
+      <nav style={navStyle}>
+        <Link to="/">Home</Link>
+        <Link to="/search">Search</Link>
+        <Link to="/movie/123">Details</Link>
+      </nav>
+
+      {/* Routes */}
+      <Routes>
+        <Route path="/" element={<Home />} />
```

Comment:
Consider using React.lazy and Suspense for page components to enable lazy loading.  
This improves performance by only loading each page when its route is visited, reducing the initial bundle size.